pulsar-broker/src/test/java/org/apache/pulsar/client/api/RetryTopicTest.java

```diff
@@ -128,6 +133,100 @@ public void testRetryTopic() throws Exception {
         checkConsumer.close();
     }
 
+    @Data
+    public static class Foo {
+        @Nullable
+        private String field1;
+        @Nullable
+        private String field2;
+    }
+
+    @Data
+    public static class FooV2 {
+        @Nullable
+        private String field1;
+        @Nullable
+        private String field2;
+        @Nullable
+        private String field3;
+    }
+
+    @Test(timeOut = 20000)
+    public void testAutoConsumeSchemaRetryLetter() throws Exception {
+        final String topic = "persistent://my-property/my-ns/retry-letter-topic";
+        final String subName = "my-subscription";
+        final int maxRedeliveryCount = 1;
+        final int sendMessages = 10;
+
+        admin.topics().createNonPartitionedTopic(topic);
+
+        @Cleanup
+        PulsarClient newPulsarClient = newPulsarClient(lookupUrl.toString(), 0);// Creates new client connection
+        Consumer<FooV2> deadLetterConsumer = newPulsarClient.newConsumer(Schema.AVRO(FooV2.class))
```

Comment:
why we should use `deadLetterConsumer`?

Reply:
I follow the test pattern from `testRetryTopic` and `testAutoConsumeSchemaDeadLetter`. Even `testRetryTopic` set up the DLQ.

May you directly submit a patch onto this pull request?